### ISSUE TITLE
Bug: v6 `typeMatchesSpecifier` `LibSpecifier` should return true for intrinsic types (string, number, boolean, etc)

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#6867]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
typeMatchesSpecifier(
  type,
  {
    from: "lib",
    name: "string",
  },
  parserServices.program)
```

### Expected Result

Native types should be considered part of "lib"

### Actual Result

Native types aren't match my any of the object specifiers.

### Additional Info

This works:

```ts
typeMatchesSpecifier(
  type,
  "string",
  parserServices.program)
```

So maybe this change isn't needed.
Though I'd expect every type to able to be specified with one of the object specifiers.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.0.0-alpha.121` |
| `@typescript-eslint/parser`            | `6.0.0-alpha.121` |
| `@typescript-eslint/scope-manager`     | `6.0.0-alpha.121` |
| `@typescript-eslint/typescript-estree` | `6.0.0-alpha.121` |
| `@typescript-eslint/type-utils`        | `6.0.0-alpha.121` |
| `@typescript-eslint/utils`             | `6.0.0-alpha.121` |
| `TypeScript`                           | `5.0.3` |
| `ESLint`                               | `8.36.0` |
| `node`                                 | `18.15.0` |
